datasets/us/navy/crawler.py

```diff
@@ -0,0 +1,148 @@
+from lxml import html, etree
+from urllib.parse import urljoin
+import orjson
+import re
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+from zavod.shed.zyte_api import fetch_html
+
+BASE_URL = "https://www.navy.mil"
+API_URL = "https://www.navy.mil/API/ArticleCS/Public/GetList?moduleID=709&dpage=%d&TabId=119&language=en-US"
+
+TITLE_REGEX = re.compile(
+    r"^(.*)\b(Admiral|Adm\.)\s+(.*)$"
+)  # Regular expression to match either "Admiral" or "Adm."
+
+
+def emit_person(
+    context: Context, country: str, source_url: str, role: str, name: str, title: str
+):
+    person = context.make("Person")
+    person.id = context.make_id(country, name, role)
+    person.add("name", name)
+    person.add("position", role)
+    person.add("sourceUrl", source_url)
+    person.add("title", title)
+
+    position = h.make_position(context, role, country=country, topics=["gov.security"])
+
+    categorisation = categorise(context, position, is_pep=True)
+    if categorisation.is_pep:
+        occupancy = h.make_occupancy(context, person, position)
+        if occupancy:
+            context.emit(person, target=True)
+            context.emit(position)
+            context.emit(occupancy)
+
+
+def crawl_person(context: Context, item_html: str) -> None:
+    doc = html.fromstring(item_html)
+    link_el = doc.find(".//a")
+    if link_el is None:
+        return
+    url = link_el.get("href")
+    name = link_el.find(".//h2").text_content().strip()
+    role = link_el.find(".//h3").text_content().strip()
+
+    match = re.search(TITLE_REGEX, name)
+    if match:
+        # Extract the full title (including everything before "Admiral" or "Adm.")
+        title = match.group(1).strip() + " " + match.group(2).strip()
+        # Extract the name (everything after "Admiral" or "Adm.")
+        name = match.group(3).strip()
+    else:
+        # Log a warning if no title is found
+        context.log.warn(f"Failed to extract title from name: {name}")
```

Comment:
looks like we can make this info level - I see there are legitimate cases without titles

```suggestion
        context.log.info(f"Failed to extract title from name: {name}")
```